### DISCUSSION TITLE
share data between tables?

### DISCUSSION THREAD

**Discussion by shivtorov** [EvotecIT/PSWriteHTML#405] [Q&A]
**Is there a way to use one data source for multiple tables?**
I.e. I have one large data table, and want to show it multiple times without increasing HTML file size.

My scenario:
1) I have one large table with thousands of ActiveDirectory users (`$users = Get-ADUser`)
2) I want to produce HTML file that has multiple tabs, each showing only subset of the original `$users` data table, e.g.:
- disabled users
- expired users
- users with mailboxes
- users from a particuler OU1
- users from a particuler OU2
- all users

Using the following code, each New-HtmlTable saves its own copy of data into HTML file.
So, in my scenario described above, the resulting HTML file grows very fast when I add additional tabs

```powershell
# table with 10'000 rows and 11 columns
$table = @(
    (1..10000).ForEach{ $row = [ordered]@{Id=$_}; (1..10).ForEach{ $row.Add("A$_", 'x' * $_) }; [PSCustomObject]$row }
)

# if we display the same table twice, the html file size doubles
New-HTML -Online -FilePath 'D:\Temp\PSWriteHTML\3.html' {
    New-HTMLTableOption -DataStore JavaScript

    New-HTMLSection -HeaderText 'Table 1 (full)' -CanCollapse -Collapsed {
        New-HTMLTable -Table $table
    }
    New-HTMLSection -HeaderText 'Table 2 (the same one)' -CanCollapse -Collapsed {
        New-HTMLTable -Table $table
    }
}

# in this example, each HTMLTable saves its own copy of data as well
New-HTML -Online -FilePath 'D:\Temp\PSWriteHTML\4.html' {
    New-HTMLTableOption -DataStore JavaScript

    New-HTMLSection -HeaderText 'Table 1 (full)' -CanCollapse -Collapsed {
        New-HTMLTable -Table $table
    }
    New-HTMLSection -HeaderText 'Table 2 (1st half)' -CanCollapse -Collapsed {
        New-HTMLTable -Table ($table.Where{ $_.Id -lt 5000 })
    }
    New-HTMLSection -HeaderText 'Table 3 (2nd half)' -CanCollapse -Collapsed {
        New-HTMLTable -Table ($table.Where{ $_.Id -ge 5000 })
    }
}
```

**Comment by PrzemyslawKlys** [comment]
Currently it's not possible, but was/is on my radar. The problem with shared data is that you need to define it once, and then link it to all the tables. But once you link it to all the tables, you need to actually limit output by using JavaScript. That means I need to somehow provide an easy way to filter the table based on some conditions.
